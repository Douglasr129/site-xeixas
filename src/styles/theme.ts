/* eslint-disable quotes */
export const theme = {
	colors: {
		primaryColor: '#0a1128',
		secondaryColor: '#dc143c',
		white: '#FFFFFF',
		mediumGray: '#DDDDDD',
		darkColor: '#2d2d2d',
	},
	font: {
		family: {
			default: "'Open Sans', sans-serif",
			secundary: "'Montserrat',sans-serif",
		},
		sizes: {
			xsmall: '0.8rem',
			small: '1.6rem',
			medium: '2.4rem',
			large: '3.2rem',
			xlarge: '4.0rem',
			xxlarge: '4.8rem',
			huge: '5.6rem',
			xhuge: '6.4rem',
		},
	},
	media: {
		lteMedium: '(max-width: 768px)',
	},
	spacing: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
};
