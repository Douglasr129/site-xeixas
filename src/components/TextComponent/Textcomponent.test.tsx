import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { TextComponent } from '.';
import { theme } from '../../styles/theme';

describe('<TextComponent />', () => {
	it('should render a TextComponent', () => {
		renderTheme(<TextComponent Text="Children"></TextComponent>);
		expect(screen.getByText('Children')).toBeInTheDocument();
	});
	it('should rende elemnets HTML', () => {
		const html = '<h1>Test<h1/>';
		renderTheme(<TextComponent Text={html}></TextComponent>);
		expect(screen.getByRole('heading', { name: 'Test' })).toBeInTheDocument();
	});
	it('', () => {
		const { container } = renderTheme(
			<TextComponent darkMode={true} Text="Children"></TextComponent>,
		);
		expect(container.firstChild).toHaveStyleRule('color', theme.colors.white);
	});
	it('', () => {
		const { container } = renderTheme(
			<TextComponent darkMode={true} Text="Children"></TextComponent>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
