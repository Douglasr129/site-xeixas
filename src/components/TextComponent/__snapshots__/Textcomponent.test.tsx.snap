// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent />  1`] = `
.c0 {
  color: #FFFFFF;
}

<div
  class="c0"
>
  Children
</div>
`;
